### DISCUSSION TITLE
is it possible to select an existing object by id?

### DISCUSSION THREAD

**Discussion by r03ert0** [spakin/SimpInkScr#118] [Q&A] [answered]
Thank you for the great extension!
I would like to do a boolean operation between a list of generated objects and an existing object. Using the SVG editor in Inkscape I added an "id" to the object I would like to use. Is it possible to select that object by id and assign it to a variable?

**Comment by r03ert0** [comment]
found it!

does this sound ok?

```python
obj = None
for r in all_shapes():
    if r._inkscape_obj.eid == 'obj':
        obj = r
        break
if obj:
    obj.style(fill='red')
```

**Comment by spakin** [comment, marked as answer]
Yes, that sounds ok.  Thanks for posting your solution.

I would recommend, however, using [`get_inkex_object()` ](https://github.com/spakin/SimpInkScr/wiki/Advanced-usage#arbitrary-svg-objects) instead of `_inkscape_obj`.  As the leading underscore implies, `_inkscape_obj` is intended for Simple Inkscape Scripting internal use, not for user scripts.  Also, `_inkscape_obj` is defined only on Simple Inkscape Scripting _shapes_ while `get_inkex_object()` is defined on all Simple Inkscape Scripting objects, including for example filters and gradients.

Here's an alternative expression of your code using list comprehensions, of which I'm quite fond:
```Python
try:
    obj = [r for r in all_shapes() if r.get_inkex_object().eid == 'obj'][0]
    obj.style(fill='red')
except IndexError:
    pass
```

**Reply by hyperneuron** [reply]
Hey there, this topic is exactly what I needed. I am trying to fill an object with a gradient like shown in the gradient example. But the gradient will not be visible until I change the gradient direction with the gradient tool. Is there any way to refresh the dom and reload after changing things like this?

**Reply by spakin** [reply]
Do you think the problem may be related to Inkscape bug [#2956](https://gitlab.com/inkscape/inkscape/-/issues/2956)?

**Reply by hyperneuron** [reply]
I couldnt reproduce it with the file posted in this error report. I am using Inkscape 1.3.2
But when I apply my script to the existing rectangle which was created by hand before I get the exact effect. The rectangle changes its color to the last color of the gradient. But when I move it the gradient gets visible. So it could be related.
I tried to move the rectangle with the script to apply the gradient. But this is also not working.
Here is my script:

```
colorful = linear_gradient((0, 0), (1, 1))
colorful.add_stop(0.00, 'cyan')
colorful.add_stop(0.50, 'red')
colorful.add_stop(1.00, 'green')

l2r=linear_gradient((0, 0), (1, 0),template=colorful)

for r in all_shapes():
    if r.get_inkex_object().eid == "rect1":
        r.style(fill=l2r)
```

**Reply by spakin** [reply]
I can reproduce the behavior you're seeing with your script.  I believe this is indeed another case of [#2956](https://gitlab.com/inkscape/inkscape/-/issues/2956) (or something very similar).  As you can see from the issue page, at the time of this writing the issue is still open, but the Inkscape team has identified the code change that introduced the bad behavior.  We now need to wait for someone who knows what he's doing to fix the Inkscape bug.

**Reply by hyperneuron** [reply]
@spakin thanks for your fast reply. Its open since 2 years :( . So can take some more time. Thanks for pointing me to the bug report. since I really need this gradient feature I will try to find a workaround or might have to use an other lib for now.
